### PR TITLE
feat: add panic sentry and leader election metric

### PR DESCRIPTION
**Pull request by Han-Ya-Jun**
### Description
add panic sentry and leader election metric

Fixes # (issue)

### Checklist

- [x] 填写 PR 描述及相关 issue (write PR description and related issue)
- [x] 代码风格检查通过 (code style check passed)
- [ ] PR 中包含单元测试 (include unit test)
- [x] 单元测试通过 (unit test passed)
- [x] 本地开发联调环境验证通过 (local development environment verification passed)

### CODE REVIEW

**Review by alex-smile — APPROVED** [review]

**Comment by alex-smile** [comment]
LGMT

**Review by wklken — APPROVED** [review]
LGTM

File: Makefile [hunk 1]
```diff
@@ -106,7 +106,7 @@ build-common: $(BUILD_PATH) generate manifests fmt vet
 
 .PHONY: build
 build: ## Build manager binary.
-	go build -ldflags "-X github.com/TencentBlueKing/blueking-apigateway-operator/pkg/version.Version=${VERSION}  \
+	go build -ldflags "-X github.com/TencentBlueKing/blueking-apigateway-operator/pkg/version.Version=`git describe --tags --abbrev=0`  \
```
Comment (wklken) [inline Makefile:109]: 如果 1.13 已经打tag, 此时发1.12的patch包, 会拿到最新的tag? 导致版本错误?

Comment (Han-Ya-Jun) [reply Makefile:109]: Version: v1.0.0
Commit: a2ce81f410778a8b5dea582978b49128483b17cd
Build Time: 2023-11-03_10:28:43
Go Version: go version go1.20.4 linux/amd64

测试了一下ok，`git describe --tags --abbrev=0` 这个是获取当前版本代码的最新tag，由于我们是基于tag来发布就不会有问题，如果是基于分支就会有问题

**Pull request merged**